streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -77,27 +316,127 @@
 
     private boolean rebalanceInProgress = false;  // if we are in the middle of a rebalance, it is not safe to commit
 
+    // for testing only
```

Comment:
I added this constructor to allow mocking both task creators -- otherwise `TaskManagerTest` would require complex mocking and major rewrite

Reply:
Not sure I follow here: can we just use easy-mock for the creators here, what are the actual issues that would cause major re-write in the test class?

Reply:
If we cannot mock both `TaskCreators`, we need to mock the `InternalTopologyBuilder` what is much more complicated -- I tried this originally, but it was quite a mess. Also note, that `TaskManagerTest` makes a lot of assertions what method the `TaskManager` calls on the created tasks -- however, we would not be able to easily check this if we don't pass in mocked tasks, and we cannot pass in mock tasks without mocking the `TaskCreator`.

If you think this is too much of a hack, I would recommend to get rid of this constructor in a follow up PR and rewrite the `TaskManagerTest` their (to keep this PR focused and scoped).